tools/kata-agent-ctl/src/client.rs

```diff
@@ -0,0 +1,1193 @@
+        st: ServiceType::Agent,
+        fp: agent_cmd_container_stats,
+    },
+    AgentCmd {
+        name: "StopTracing",
+        st: ServiceType::Agent,
+        fp: agent_cmd_sandbox_tracing_stop,
+    },
+    AgentCmd {
+        name: "UpdateInterface",
+        st: ServiceType::Agent,
+        fp: agent_cmd_sandbox_update_interface,
+    },
+    AgentCmd {
+        name: "UpdateRoutes",
+        st: ServiceType::Agent,
+        fp: agent_cmd_sandbox_update_routes,
+    },
+    AgentCmd {
+        name: "WaitProcess",
+        st: ServiceType::Agent,
+        fp: agent_cmd_container_wait_process,
+    },
+];
+
+static BUILTIN_CMDS: &'static [BuiltinCmd] = &[
+    BuiltinCmd {
+        name: "echo",
+        descr: "Display the arguments",
+        fp: builtin_cmd_echo,
+    },
+    BuiltinCmd {
+        name: "help",
+        descr: "Alias for 'list'",
+        fp: builtin_cmd_list,
+    },
+    BuiltinCmd {
+        name: "list",
+        descr: "List all available commands",
+        fp: builtin_cmd_list,
+    },
+    BuiltinCmd {
+        name: "repeat",
+        descr: "Repeat the next command 'n' times [-1 for forever]",
+        fp: builtin_cmd_repeat,
+    },
+    BuiltinCmd {
+        name: "sleep",
+        descr:
+            "Pause for specified period number of nanoseconds (supports human-readable suffixes [no floating points numbers])",
+        fp: builtin_cmd_sleep,
+    },
+    BuiltinCmd {
+        name: CMD_QUIT,
+        descr: "Exit this program",
+        fp: builtin_cmd_quit,
+    },
+];
+
+fn get_agent_cmd_names() -> Vec<String> {
+    let mut names = Vec::new();
+
+    for cmd in AGENT_CMDS {
+        names.push(cmd.name.to_string());
+    }
+
+    names
+}
+
+fn get_agent_cmd_details() -> Vec<String> {
+    let mut cmds = Vec::new();
+
+    for cmd in AGENT_CMDS {
+        let service = match cmd.st {
+            ServiceType::Agent => "agent",
+            ServiceType::Health => "health",
+        };
+
+        cmds.push(format!("{} ({} service)", cmd.name, service));
+    }
+
+    cmds
+}
+
+fn get_agent_cmd_func(name: &str) -> Result<AgentCmdFp> {
+    for cmd in AGENT_CMDS {
+        if cmd.name == name {
+            return Ok(cmd.fp);
+        }
+    }
+
+    Err(format!("Invalid command: {:?}", name))
+}
+
+fn get_builtin_cmd_details() -> Vec<String> {
+    let mut cmds = Vec::new();
+
+    for cmd in BUILTIN_CMDS {
+        cmds.push(format!("{} ({})", cmd.name, cmd.descr));
+    }
+
+    cmds
+}
+
+fn get_all_cmd_details() -> Vec<String> {
+    let mut cmds = get_builtin_cmd_details();
+
+    cmds.append(&mut get_agent_cmd_names());
+
+    cmds
+}
+
+fn get_builtin_cmd_func(name: &str) -> Result<BuiltinCmdFp> {
+    for cmd in BUILTIN_CMDS {
+        if cmd.name == name {
+            return Ok(cmd.fp);
+        }
+    }
+
+    Err(format!("Invalid command: {:?}", name))
+}
+
+// The vsock crate doesn't work ;((
+fn client_create_vsock_fd_with_nix_crate(cid: libc::c_uint, port: u32) -> Result<RawFd> {
```

Comment:
I think you should remove `_with_nix_crate` from the function name.

Reply:
Done.